gamedata/alldefs_post.lua

```diff
@@ -857,6 +857,11 @@ function UnitDef_Post(name, uDef)
 		end
 	end
 
+	-- Shield Rework
+	if modOptions.shieldsrework == true and uDef.weapondefs and uDef.weapondefs.repulsor then
+		uDef.onoffable = true
+	end
```

Comment:
This is checking for a `repulsor` key in the table of the unit's weapons, not on the weapon itself.
```suggestion
	if modOptions.shieldsrework == true then
		for weaponDefName, weaponDef in pairs(uDef.weapondefs) do
			if weapondefsrepulsor then
				uDef.onoffable = true
			end
		end
	end
```

Reply:
Not sure what "weapondefsrepulsor" is shorthand for in your example. From my examination of the unitdefs, a `repulsor` key in the weapondefs seemed like enough indication that a unit has a shield. I can open up the weapondefs and check for a `shield` key with the `repulser` set to True instead, is this what you had in mind? 


```suggestion
	if modOptions.shieldsrework == true and uDef.weapondefs then
		for _, weapon in pairs(uDef.weapondefs) do
			if weapon.shield and weapon.shield.repulser then
				uDef.onoffable = true
			end
		end
	end

```

Reply:
Whoops, good catch, you are correct

Reply:
I implemented this change. I also squashed my commits together instead of creating a fourth, hopefully that doesn't mess with the review process.